main.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+import os
+import pandas as pd
+
+from helpers.config_helper import load_env_file
+from helpers.log_helper import create_log
+from helpers.obfuscation_helper import obfuscate
+from helpers.query_helper import build_new_patrons_query
+from lib import (AvroEncoder, DbClient, DbMode, GeocoderApiClient,
+                 KinesisClient, S3Client)
+
+_EST_TIMEZONE = datetime.timezone(datetime.timedelta(days=-1, seconds=68400))
+
+
+def main():
+    load_env_file(os.environ['ENVIRONMENT'], 'config/{}.yaml')
+    logger = create_log(__name__)
+
+    # Set up all the clients
+    sierra_client = DbClient(DbMode.SIERRA)
+    geocoder_client = GeocoderApiClient()
+    avro_encoder = AvroEncoder()
+    kinesis_client = KinesisClient()
+    s3_client = S3Client()
+
+    has_max_batches = 'MAX_BATCHES' in os.environ
+    batch_number = 1
+    finished = False
+    while not finished:
+        # Retrieve the state if necessary either from the cache or the config
+        if (os.environ.get('IGNORE_CACHE', False) == 'True'
+                and batch_number == 1):
+            poller_state = {'creation_dt':
+                            os.environ.get('STARTING_CREATION_DT',
+                                           '2020-01-01 00:00:00-05'),
+                            'update_dt':
+                            os.environ.get('STARTING_UPDATE_DT',
+                                           '2020-01-01 00:00:00-05'),
+                            'deletion_date':
+                            os.environ.get('STARTING_DELETION_DATE',
+                                           '2020-01-01')}
+        elif os.environ.get('IGNORE_CACHE', False) != 'True':
+            poller_state = s3_client.fetch_state()
+
+        # Get data from Sierra
```

Comment:
This core functionality is great and very clear as to what each section is doing. Just wondering if it would make sense to pull this chunk of code out into its own module/class.

This is probably just personal preference but I like to use `main.py` to orchestrate everything and then have this kind of stuff live separately.

Reply:
Moved to a helper method. I think it should live in main, though, because it's iteration and environment-dependent, whereas all the other classes are a little more independent of each other and when/how they're used.

Reply:
oh shoot, sorry! I actually meant everything in the `while not finished` loop (with the exception of the loop exit logic), but that definitely wasn't clear.

I agree that the organizational approach you've taken is clear and neatly delineates between "core" logic and classes that perform specific actions. I just thought it might be cleaner to also delineate between the logic around the central while loop and the logic that runs for each iteration. Though perhaps that's taking things too far?

Reply:
Ah I see! In the future most of the main function will be moved to a check_new_patrons method and there will be check_updated_patrons and check_deleted_patrons methods as well, so this is temporary. I think it's fine to keep it this way for now and make that change later, though, if it's ok with you.

Reply:
that's fine with me! sounds like a good plan and good reason to leave this as-is for now